### PR TITLE
fix(reports): automatically aggregate when groupby is set in LinePlot and BarPlot

### PR DESCRIPTION
**Pull request by luisbergua**
### Fixes:
* https://wandb.atlassian.net/browse/WB-12407
* https://wandb.atlassian.net/browse/WB-16436
### Description:
* Updated both `LinePlot` and `BarPlot` to:
   * Automatically set `aggregate=True` if `groupby` is set. It was confusing for users that setting `groupby` didn't make any difference on the UI and they had to also set `aggregate`
   * If `groupby` isn't set, allow users to set `aggregate` (If True runs are grouped by None as in the UI)
   * Added `_metric_to_backend_groupby` and `_metric_to_frontend_groupby` to handle groupby values and have more flexibility:
       * Before users could only use `epochs.value` or `keys.value.key1` for nested config keys, making it hard for them to figure that out
       * Now they can simply use `epochs`, `keys.key1` or `wr.Config('epochs')` or `wr.Config('keys.key1')` which is more intuitive and aligns with the rest of the code
### Testing:
Added comprehensive tests for the new groupby functionality:

* **`test_metric_to_backend_groupby()`** - Tests the `_metric_to_backend_groupby` function with various input formats:
  - `Config("epochs")` → `"epochs.value"`
  - `Config("keys.key1")` → `"keys.value.key1"`
  - `"config.epochs"` → `"epochs.value"`
  - `"epochs"` → `"epochs.value"` (legacy support)
  - Edge cases (None, empty string)

* **`test_metric_to_frontend_groupby()`** - Tests the `_metric_to_frontend_groupby` function for proper reverse conversion:
  - `"epochs.value"` → `Config("epochs")`
  - `"keys.value.key1"` → `Config("keys.key1")`
  - Non-config paths pass through unchanged

* **`test_groupby_aggregate_behavior()`** - Validates the auto-aggregate feature:
  - Panels with `groupby` automatically set `aggregate=True`
  - Panels without `groupby` allow manual `aggregate` control
  - Tests both LinePlot and BarPlot with Config objects and strings

Before:
image

````
import wandb_workspaces.reports.v2 as wr

entity = "luis_team_test"
project = "group_line_bar_plot_2"

report = wr.Report(
    entity=entity,
    project=project,
    width="fluid",
    blocks=[
        wr.PanelGrid(
            panels=[
                wr.LinePlot(
                    x="_step",
                    y=["metric"],
                    groupby="epochs.value",
                    layout=wr.Layout(w=10, h=6),
                    aggregate=True,
                ),
                wr.BarPlot(
                    metrics=["bar_metric"],
                    groupby="epochs",
                    layout=wr.Layout(w=10, h=6),
                ),
                wr.LinePlot(
                    x="_step",
                    y=["metric"],
                    groupby="keys.value.key1",
                    layout=wr.Layout(w=10, h=6),
                    aggregate=True,
                ),
                wr.BarPlot(
                    metrics=["bar_metric"],
                    groupby="keys.key1",
                    layout=wr.Layout(w=10, h=6),
                ),
            ]
        )
    ]
)
report.save()

url = report.url


new_report = wr.Report.from_url(url)
new_report.save()
````
After:
image

````
import wandb_workspaces.reports.v2 as wr

entity = "luis_team_test"
project = "group_line_bar_plot_2"

report = wr.Report(
    entity=entity,
    project=project,
    width="fluid",
    blocks=[
        wr.PanelGrid(
            panels=[
                wr.LinePlot(
                    x="_step",
                    y=["metric"],
                    groupby=wr.Config("epochs"),
                    layout=wr.Layout(w=10, h=6),
                ),
                wr.BarPlot(
                    metrics=["bar_metric"],
                    groupby="epochs",
                    layout=wr.Layout(w=10, h=6),
                ),
                wr.LinePlot(
                    x="_step",
                    y=["metric"],
                    groupby=wr.Config("keys.key1"),
                    layout=wr.Layout(w=10, h=6),
                ),
                wr.BarPlot(
                    metrics=["bar_metric"],
                    groupby="keys.key1",
                    layout=wr.Layout(w=10, h=6),
                ),
            ]
        )
    ]
)
report.save()

url = report.url


new_report = wr.Report.from_url(url)
new_report.save()
````

### CODE REVIEW

File: wandb_workspaces/reports/v2/interface.py [hunk 1]
```diff
@@ -3602,6 +3606,61 @@ def _metric_to_frontend_panel_grid(x: str):
...
+
+    # 2) drop an explicit "config." prefix for uniform handling
+    if val.startswith("config."):
+        val = val.split("config.", 1)[1]
+
+    segments = val.split(".")
+
+    # 3) if we already have ".value" immediately after the first segment, keep it
+    if len(segments) >= 2 and segments[1] == "value":
+        return val
```
[the remark this answers is not in the document]
Comment (jo-fang) [reply wandb_workspaces/reports/v2/interface.py:3638]: Ah okay and if `value` is not in the correct position , we can help user out?

https://weightsandbiases.slack.com/archives/C08ES2XLFSA/p1750958359384939?thread_ts=1750330523.444509&cid=C08ES2XLFSA

[the remark this answers is not in the document]
Comment (luisbergua) [reply wandb_workspaces/reports/v2/interface.py:3638]: We discussed internally, `value` might also be a key added for the user so don't wanna block that case, leaving as it is for now and can improve in the future if we have specific cases that aren't handled well
